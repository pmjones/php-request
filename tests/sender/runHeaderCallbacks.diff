001+ 
001- foo: bar
002- baz: dib, zim
003+ Fatal error: Uncaught Error: Call to undefined method SapiResponse::addHeaderCallback() in /Users/pmjones/Code/pmjones/php-request/tests/sender/runHeaderCallbacks.php:3
004+ Stack trace:
005+ #0 {main}
006+   thrown in /Users/pmjones/Code/pmjones/php-request/tests/sender/runHeaderCallbacks.php on line 3
004- array(2) {
005-   [0]=>
006-   string(8) "foo: bar"
007-   [1]=>
008-   string(13) "baz: dib, zim"
009- }