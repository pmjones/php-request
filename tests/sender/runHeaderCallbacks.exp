foo: bar
baz: dib, zim
--HEADERS--
array(2) {
  [0]=>
  string(8) "foo: bar"
  [1]=>
  string(13) "baz: dib, zim"
}