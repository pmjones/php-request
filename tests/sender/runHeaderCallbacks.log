
---- EXPECTED OUTPUT
foo: bar
baz: dib, zim
--HEADERS--
array(2) {
  [0]=>
  string(8) "foo: bar"
  [1]=>
  string(13) "baz: dib, zim"
}
---- ACTUAL OUTPUT

--HEADERS--
Fatal error: Uncaught Error: Call to undefined method SapiResponse::addHeaderCallback() in /Users/pmjones/Code/pmjones/php-request/tests/sender/runHeaderCallbacks.php:3
Stack trace:
#0 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/tests/sender/runHeaderCallbacks.php on line 3
---- FAILED
