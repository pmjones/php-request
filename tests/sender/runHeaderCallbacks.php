<?php
$response = new SapiResponse();
$response->addHeaderCallback(function($response) {
    $response->setHeader('Foo', 'bar');
    $response->setHeader('Baz', 'dib');
    $response->addHeader('Baz', 'zim');
});
(new SapiResponseSender())->send($response);
var_dump(headers_list());
// it appears EXPECTHEADERS can't handle duplicate headers
