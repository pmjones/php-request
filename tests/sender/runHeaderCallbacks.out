
--HEADERS--
Fatal error: Uncaught Error: Call to undefined method SapiResponse::addHeaderCallback() in /Users/pmjones/Code/pmjones/php-request/tests/sender/runHeaderCallbacks.php:3
Stack trace:
#0 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/tests/sender/runHeaderCallbacks.php on line 3